pvlib/iotools/tmy.py

```diff
@@ -198,9 +224,17 @@ def read_tmy3(filename, coerce_year=None, recolumn=True):
     # NOTE: as of pvlib-0.6.3, min req is pandas-0.18.1, so pd.to_timedelta
     # unit must be in (D,h,m,s,ms,us,ns), but pandas>=0.24 allows unit='hour'
     data.index = data_ymd + pd.to_timedelta(shifted_hour, unit='h')
-
-    if recolumn:
-        data = _recolumn(data)  # rename to standard column names
+    if map_variables is not True:
+        if recolumn:
+            data = _recolumn(data)  # rename to standard column names
+        warnings.warn(
+            'TMY3 variable names will be renamed to pvlib conventions by '
+            'default starting in pvlib 0.11.0. Specify map_variables=True '
+            'to enable that behavior now, or specify map_variables=False '
+            'to hide this warning.', pvlibDeprecationWarning)
```

Comment:
Taking another look now, this warning is now emitted even if `map_variables=False`, which of course is not desirable.  I think this warning should go in a separate `if map_variables is None:` branch instead of being inside this one.